File: My_Notes/Programming_Note/Samples/async_samples/go/async1.go
package main

import "sync"

func asyncFunc1() {
	job := job{
		db:  3,
		api: 1,
		ui:  2,
	}
	var dbEngineer dbEngineer
	var apiEngineer apiEngineer
	var uiEngineer uiEngineer

	wg := sync.WaitGroup{}
	wg.Add(3)

	go func() {
		dbEngineer.do(job)
		wg.Done()
	}()
	go func() {
		apiEngineer.do(job)
		wg.Done()
	}()
	go func() {
		uiEngineer.do(job)
		wg.Done()
	}()

	wg.Wait()
}

File: My_Notes/Programming_Note/Samples/async_samples/ts/async1.ts
function async1():Promise<void>{
    var job = new Job(3,1,2);
    var dbEngineer = new DBEngineer();
    var apiEngineer = new APIEngineer();
    var uiEngineer = new UIEngineer();

    var task1:Promise<void> = dbEngineer.do(job)
    var task2:Promise<void> = apiEngineer.do(job)
    var task3:Promise<void> = uiEngineer.do(job)

    Promise.all([task1, task2, task3])
    return null;
}